### ISSUE TITLE
[BUG] AttributeError: module 'numpy' has no attribute 'complex'

### ISSUE THREAD

**Issue by pililac** [ECSHackWeek/impedance.py#236]
Today I updated my mini-forge environment. My impedance script ran well beforehand and failed afterward. I suspect the updated numpy package. Was numpy.dtype numpy.complex removed/changed?

Here is the error log from my jupyter notebook:

````
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[10], line 77
     74 bounds_CPE3_0 = [8e-3, 3.8e-1]
     75 bounds_CPE3_1 = [0.2, 0.7]
---> 77 circuit.fit(
     78     frequencies,
     79     Z,
     80     bounds=(
     81         [
     82             bounds_R0[0],
     83             bounds_R1[0],
     84             bounds_C1[0],
     85             bounds_R2[0],
     86             bounds_CPE2_0[0],
     87             bounds_CPE2_1[0],
     88             bounds_R3[0],
     89             bounds_Wo3_0[0],
     90             bounds_Wo3_1[0],
     91             bounds_CPE3_0[0],
     92             bounds_CPE3_1[0],
     93         ],
     94         [
     95             bounds_R0[1],
     96             bounds_R1[1],
     97             bounds_C1[1],
     98             bounds_R2[1],
     99             bounds_CPE2_0[1],
    100             bounds_CPE2_1[1],
    101             bounds_R3[1],
    102             bounds_Wo3_0[1],
    103             bounds_Wo3_1[1],
    104             bounds_CPE3_0[1],
    105             bounds_CPE3_1[1],
    106         ],
    107     ),
    108     # global_opt=True,
    109     # weight_by_modulus=True,
    110 )
    112 loop_results = pd.DataFrame(
    113     [
    114         {
   (...)
    118     ]
    119 )
    120 for idx, param in enumerate(circuit.get_param_names()[0]):

File C:\other_software\miniforge3\envs\data_analysis\lib\site-packages\impedance\models\circuits\circuits.py:107, in BaseCircuit.fit(self, frequencies, impedance, bounds, weight_by_modulus, **kwargs)
    105 if not isinstance(impedance, np.ndarray):
    106     raise TypeError('impedance is not of type np.ndarray')
--> 107 if impedance.dtype != np.complex:
    108     raise TypeError('impedance array should have a complex ' +
    109                     f'dtype (currently {impedance.dtype})')
    110 if len(frequencies) != len(impedance):

File C:\other_software\miniforge3\envs\data_analysis\lib\site-packages\numpy\__init__.py:284, in __getattr__(attr)
    281     from .testing import Tester
    282     return Tester
--> 284 raise AttributeError("module {!r} has no attribute "
    285                      "{!r}".format(__name__, attr))

AttributeError: module 'numpy' has no attribute 'complex'

````

As far as I understand, the script tests if the dtype of impedance is complex, but 'complex' is no valid dtype (anymore?). 
Is that an individual problem of my setup or a general incompatibility between impedance and numpy?

Here is the output of my python environments `conda list`:

<details>
  <summary>Click me</summary>
  
```conda list
# packages in environment at C:\other_software\miniforge3\envs\data_analysis:
#
# Name                    Version                   Build  Channel
_ipython_minor_entry_point 8.7.0                hb6b4a82_0    conda-forge
altair                    4.2.0                    pypi_0    pypi
anyio                     3.6.2              pyhd8ed1ab_0    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py310h8d17308_3    conda-forge
astroid                   2.12.13         py310h5588dad_0    conda-forge
asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
attrs                     22.2.0             pyh71513ae_0    conda-forge
autopep8                  1.6.0              pyhd8ed1ab_1    conda-forge
babel                     2.11.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
black                     22.10.0         py310h5588dad_2    conda-forge
bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                hcfcfb64_8    conda-forge
brotli-bin                1.0.9                hcfcfb64_8    conda-forge
brotlipy                  0.7.0           py310h8d17308_1005    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2022.12.7            h5b45459_0    conda-forge
certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h628cb3f_3    conda-forge
charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
click                     8.1.3           win_pyhd8ed1ab_2    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.1.2              pyhd8ed1ab_0    conda-forge
conda-package-handling    2.0.2              pyh38be061_0    conda-forge
conda-package-streaming   0.7.0              pyhd8ed1ab_1    conda-forge
contourpy                 1.0.6           py310h232114e_0    conda-forge
cryptography              38.0.4          py310h6e82f81_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
debugpy                   1.6.4           py310h00ffb61_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
dill                      0.3.6              pyhd8ed1ab_1    conda-forge
dnspython                 2.2.1              pyhd8ed1ab_0    conda-forge
docstring-to-markdown     0.11               pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
et_xmlfile                1.0.1                   py_1001    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
flake8                    6.0.0              pyhd8ed1ab_0    conda-forge
flit-core                 3.8.0              pyhd8ed1ab_0    conda-forge
fonttools                 4.38.0          py310h8d17308_1    conda-forge
freetype                  2.12.1               h546665d_1    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
glib                      2.74.1               h12be248_1    conda-forge
glib-tools                2.74.1               h12be248_1    conda-forge
gst-plugins-base          1.21.3               h001b923_1    conda-forge
gstreamer                 1.21.3               h6b5321d_1    conda-forge
icu                       70.1                 h0e60522_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
impedance                 1.4.1                    pypi_0    pypi
importlib-metadata        6.0.0              pyha770c72_0    conda-forge
importlib_resources       5.10.1             pyhd8ed1ab_1    conda-forge
intel-openmp              2023.0.0         h57928b3_25922    conda-forge
ipykernel                 6.19.4             pyh025b116_0    conda-forge
ipympl                    0.9.2              pyhd8ed1ab_0    conda-forge
ipython                   8.7.0              pyh08f2357_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                8.0.4              pyhd8ed1ab_0    conda-forge
isort                     5.11.4             pyhd8ed1ab_0    conda-forge
jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jpeg                      9e                   h8ffe710_2    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                4.17.3             pyhd8ed1ab_0    conda-forge
jupyter-lsp               1.5.1              pyhd8ed1ab_0    conda-forge
jupyter_client            7.4.8              pyhd8ed1ab_0    conda-forge
jupyter_core              5.1.1           py310h5588dad_0    conda-forge
jupyter_events            0.5.0              pyhd8ed1ab_1    conda-forge
jupyter_server            2.0.6              pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.4.3              pyhd8ed1ab_0    conda-forge
jupyterlab                3.5.2              pyhd8ed1ab_0    conda-forge
jupyterlab-lsp            3.10.2             pyhd8ed1ab_0    conda-forge
jupyterlab_code_formatter 1.5.3              pyhd8ed1ab_0    conda-forge
jupyterlab_execute_time   2.3.1              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
jupyterlab_server         2.17.0             pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        3.0.5              pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.4           py310h232114e_1    conda-forge
krb5                      1.20.1               heb0366b_0    conda-forge
lazy-object-proxy         1.8.0           py310h8d17308_0    conda-forge
lcms2                     2.14                 ha5c8aab_1    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              16_win64_mkl    conda-forge
libbrotlicommon           1.0.9                hcfcfb64_8    conda-forge
libbrotlidec              1.0.9                hcfcfb64_8    conda-forge
libbrotlienc              1.0.9                hcfcfb64_8    conda-forge
libcblas                  3.9.0              16_win64_mkl    conda-forge
libclang                  15.0.6          default_h77d9078_0    conda-forge
libclang13                15.0.6          default_h77d9078_0    conda-forge
libdeflate                1.14                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.74.1               he8f3873_1    conda-forge
libhwloc                  2.8.0                h039e092_1    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
liblapack                 3.9.0              16_win64_mkl    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libprotobuf               3.21.12              h12be248_0    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libsqlite                 3.40.0               hcfcfb64_0    conda-forge
libtiff                   4.5.0                hc4f729c_0    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp-base              1.2.4                h8ffe710_0    conda-forge
libxcb                    1.13              hcd874cb_1004    conda-forge
libxml2                   2.10.3               hc3477c8_0    conda-forge
libzlib                   1.2.13               hcfcfb64_4    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.1.1           py310h8d17308_2    conda-forge
matplotlib                3.6.2           py310h5588dad_0    conda-forge
matplotlib-base           3.6.2           py310h51140c5_0    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
menuinst                  1.4.19          py310h5588dad_1    conda-forge
miniforge_console_shortcut 2.0                  h57928b3_1    conda-forge
mistune                   2.0.4              pyhd8ed1ab_0    conda-forge
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy_extensions           0.4.3           py310h5588dad_6    conda-forge
mysql-common              8.0.31               hde014ed_0    conda-forge
mysql-connector-python    8.0.31          py310h9b4d4ad_2    conda-forge
mysql-libs                8.0.31               h8b0d2c3_0    conda-forge
nbclassic                 0.4.8              pyhd8ed1ab_0    conda-forge
nbclient                  0.7.2              pyhd8ed1ab_0    conda-forge
nbconvert                 7.2.7              pyhd8ed1ab_0    conda-forge
nbconvert-core            7.2.7              pyhd8ed1ab_0    conda-forge
nbconvert-pandoc          7.2.7              pyhd8ed1ab_0    conda-forge
nbformat                  5.7.1              pyhd8ed1ab_0    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
nodejs                    18.12.1              h57928b3_0    conda-forge
notebook                  6.5.2              pyha770c72_1    conda-forge
notebook-shim             0.2.2              pyhd8ed1ab_0    conda-forge
numpy                     1.24.1          py310h4a8f9c9_0    conda-forge
openjpeg                  2.5.0                ha2aaf27_2    conda-forge
openpyxl                  3.0.10          py310h8d17308_2    conda-forge
openssl                   3.0.7                hcfcfb64_1    conda-forge
packaging                 22.0               pyhd8ed1ab_0    conda-forge
pandas                    1.5.2           py310h1c4a608_0    conda-forge
pandoc                    2.19.2               h57928b3_1    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pathspec                  0.10.3             pyhd8ed1ab_0    conda-forge
pcre2                     10.40                h17e33f8_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.2.0           py310hdbb7713_4    conda-forge
pip                       22.3.1             pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
platformdirs              2.6.2              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0              pyhd8ed1ab_5    conda-forge
ply                       3.11                       py_1    conda-forge
prometheus_client         0.15.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.36             pyha770c72_0    conda-forge
protobuf                  4.21.12         py310h00ffb61_0    conda-forge
psutil                    5.9.4           py310h8d17308_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pycodestyle               2.10.0             pyhd8ed1ab_0    conda-forge
pycosat                   0.6.4           py310h8d17308_1    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydocstyle                6.1.1              pyhd8ed1ab_0    conda-forge
pyflakes                  3.0.1              pyhd8ed1ab_0    conda-forge
pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
pylint                    2.15.9             pyhd8ed1ab_0    conda-forge
pyopenssl                 23.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.15.7          py310h1fd54f2_2    conda-forge
pyqt5-sip                 12.11.0         py310h00ffb61_2    conda-forge
pyrsistent                0.19.2          py310h8d17308_0    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.10.8          h4de0772_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.4              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.7.0                hd8ed1ab_0    conda-forge
python-lsp-server-base    1.7.0              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    3_cp310    conda-forge
pytoolconfig              1.2.4              pyhd8ed1ab_1    conda-forge
pytz                      2022.7             pyhd8ed1ab_0    conda-forge
pywin32                   304             py310h00ffb61_2    conda-forge
pywinpty                  2.0.9           py310h00ffb61_0    conda-forge
pyyaml                    6.0             py310h8d17308_5    conda-forge
pyzmq                     24.0.1          py310hcd737a0_1    conda-forge
qt-main                   5.15.6               h9580fe5_5    conda-forge
requests                  2.28.1             pyhd8ed1ab_1    conda-forge
rope                      1.6.0              pyhd8ed1ab_0    conda-forge
ruamel_yaml               0.15.80         py310h8d17308_1008    conda-forge
scipy                     1.9.1                    pypi_0    pypi
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                65.6.3             pyhd8ed1ab_0    conda-forge
sip                       6.7.5           py310h00ffb61_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
tbb                       2021.7.0             h91493d7_1    conda-forge
terminado                 0.15.0          py310h5588dad_0    conda-forge
tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h8ffe710_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomlkit                   0.11.6             pyha770c72_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.2             py310h8d17308_1    conda-forge
tqdm                      4.64.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.8.0              pyhd8ed1ab_0    conda-forge
typing                    3.10.0.0           pyhd8ed1ab_0    conda-forge
typing-extensions         4.4.0                hd8ed1ab_0    conda-forge
typing_extensions         4.4.0              pyha770c72_0    conda-forge
tzdata                    2022g                h191b570_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
ujson                     5.5.0           py310h00ffb61_1    conda-forge
unicodedata2              15.0.0          py310h8d17308_0    conda-forge
urllib3                   1.26.13            pyhd8ed1ab_0    conda-forge
vc                        14.3                 h3d8a991_9    conda-forge
vs2015_runtime            14.32.31332          h1d6e394_9    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.4.2              pyhd8ed1ab_0    conda-forge
whatthepatch              1.0.3              pyhd8ed1ab_0    conda-forge
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
widgetsnbextension        4.0.5              pyhd8ed1ab_0    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
winpty                    0.4.3                         4    conda-forge
wrapt                     1.14.1          py310h8d17308_1    conda-forge
xlsxwriter                3.0.5              pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.9                hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
yapf                      0.32.0             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
zipp                      3.11.0             pyhd8ed1ab_0    conda-forge
zstandard                 0.19.0          py310h0009e47_1    conda-forge
zstd                      1.5.2                h7755175_4    conda-forge


</details>

````

**Comment by oslopanda** [comment]
It is because that the channel release is not updated (the code here is updated)
So, one solution before the release updated:
go to your directory:
C:\other_software\miniforge3\envs\data_analysis\lib\site-packages\impedance\models\circuits\circuits.py
from line 107
Comment out the lines 107, 108 and 109:
# if impedance.dtype != np.complex:
 #raise TypeError('impedance array should have a complex ' +
# f'dtype (currently {impedance.dtype})')

save and it will temperately fix your problem.

or you could copy the code from here to your circuits.py.

**Comment by mdmurbach** [comment]
Thanks for reaching out! Yeah, as @oslopanda mentioned in #235 the issue here was that the fixed version never got updated on PyPI. I've just released [v1.4.2](https://pypi.org/project/impedance/1.4.2/) and you should be able to update (`pip install impedance --upgrade`) and it should work without the workaround.

@all-contributors please add @pililac for bug report
